pkg/server/endpoints/authorized_entryfetcher_attested_nodes.go

```diff
@@ -24,227 +23,234 @@ type attestedNodes struct {
-		firstEventTime:          firstEventTime,
-		log:                     log,
-		metrics:                 metrics,
-		lastEventID:             lastEventID,
-		missedEvents:            missedEvents,
-		seenMissedStartupEvents: make(map[uint]struct{}),
-		sqlTransactionTimeout:   sqlTransactionTimeout,
-	}, nil
-}
 
-// updateCache Fetches all the events since the last time this function was running and updates
-// the cache with all the changes.
-func (a *attestedNodes) updateCache(ctx context.Context) error {
-	// Process events skipped over previously
-	if err := a.missedStartupEvents(ctx); err != nil {
-		a.log.WithError(err).Error("Unable to process missed startup events")
+		a.fetchNodes[event.SpiffeID] = struct{}{}
+		a.eventTracker.StopTracking(eventID)
 	}
-	a.replayMissedEvents(ctx)
+}
 
-	req := &datastore.ListAttestedNodesEventsRequest{
-		GreaterThanEventID: a.lastEventID,
+func (a *attestedNodes) scanForNewEvents(ctx context.Context) error {
+	// If we haven't seen an event, scan for all events; otherwise, scan from the last event.
+	var resp *datastore.ListAttestedNodeEventsResponse
+	var err error
+	if a.firstEventTime.IsZero() {
+		resp, err = a.ds.ListAttestedNodeEvents(ctx, &datastore.ListAttestedNodeEventsRequest{})
+	} else {
+		resp, err = a.ds.ListAttestedNodeEvents(ctx, &datastore.ListAttestedNodeEventsRequest{
+			GreaterThanEventID: a.lastEvent,
+		})
 	}
-	resp, err := a.ds.ListAttestedNodesEvents(ctx, req)
 	if err != nil {
 		return err
 	}
 
-	seenMap := map[string]struct{}{}
 	for _, event := range resp.Events {
-		// If there is a gap in the event stream, log the missed events for later processing.
-		// For example if the current event ID is 6 and the previous one was 3, events 4 and 5
-		// were skipped over and need to be queued in case they show up later.
-		// This can happen when a long running transaction allocates an event ID but a shorter transaction
-		// comes in after, allocates and commits the ID first. If a read comes in at this moment, the event id for
-		// the longer running transaction will be skipped over.
-		if !a.firstEventTime.IsZero() {
-			for i := a.lastEventID + 1; i < event.EventID; i++ {
-				a.log.WithField(telemetry.EventID, i).Info("Detected skipped attested node event")
-				a.mu.Lock()
-				a.missedEvents[i] = a.clk.Now()
-				a.mu.Unlock()
-			}
-		}
-
-		// Skip fetching entries we've already fetched this call
-		if _, seen := seenMap[event.SpiffeID]; seen {
-			a.lastEventID = event.EventID
+		// event time determines if we have seen the first event.
+		if a.firstEventTime.IsZero() {
+			a.firstEvent = event.EventID
+			a.lastEvent = event.EventID
+			a.fetchNodes[event.SpiffeID] = struct{}{}
+			a.firstEventTime = a.clk.Now()
 			continue
 		}
-		seenMap[event.SpiffeID] = struct{}{}
 
-		// Update the cache
-		if err := a.updateCacheEntry(ctx, event.SpiffeID); err != nil {
-			return err
+		// track any skipped event ids, should the appear later.
```

Comment:
```suggestion
		// track any skipped event ids, should they appear later.
```

Reply:
Fixed in the next push, in both the authorized_entryfetcher_registration_entries.go and authorized_entryfetcher_attested_nodes.go files.

Thanks for the signoff and commit link, but by me redoing it in my environment and pushing it, I don't get "not all commits" are signed errors should I ever have to update it again.

Reply:
@stevend-uber Please verify that the changes were implemented in the code push, and if they meet your satisfaction, please resolve this conversation.  It keeps the focus on the actively open items.